### PR TITLE
ability to delete uploaded files

### PR DESCRIPTION
**Pull request by duckduckcodes**

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -35,8 +35,12 @@ def get_config():
 @app.post("/config/")
 def add_config(config: Config):
     db = Database().get_session()
-    res = db.execute(text("INSERT INTO config (openai_api_key) VALUES (:api_key)"), {"api_key": config.apiKey})
-    db.commit()
+    try:
+        res = db.execute(text("INSERT INTO config (openai_api_key) VALUES (:api_key)"), {"api_key": config.apiKey})
+        db.commit()
+    except HTTPException as e:
```
Comment (grumpyp) [inline backend/main.py:41]: this would be rather a `OperationException` or something else than a `HTTPException` in my eyes.

File: backend/main.py [hunk 2]
```diff
@@ -78,6 +83,30 @@ async def upload_files(files: List[UploadFile] = File(...)):
         print(f"added {file.filename} to db")
     return {"message": "Files uploaded successfully"}
 
+import time
+
+@app.delete("/files/")
+def delete_files(file: FileToDelete):
+    print("dfgdfg")
```
Comment (grumpyp) [inline backend/main.py:90]: can go

File: backend/schemas/question.py [hunk 3]
```diff
@@ -5,6 +5,9 @@
 class Question(BaseModel):
     question: str
 
+class FileToDelete(BaseModel):
+    file: str
```
Comment (grumpyp) [inline backend/schemas/question.py:9]: whats the difference between `Document` here?

File: frontend/src/renderer/App.tsx [hunk 4]
```diff
@@ -60,20 +61,24 @@ export default function Hello() {
...
+              />
+              <Route path="/upload" element={<Upload />} />
+              <Route path="/chat" element={<Chat />} />
+              <Route path="/config" element={<Config />} />
+              <Route path="/summary" element={<Summary />} />
+            </Routes>
+            <Footer />
+          </Router>
+        </MantineProvider>
+      </Provider>
```
Comment (grumpyp) [inline frontend/src/renderer/App.tsx:81]: Why did you change something here?

File: backend/main.py (hunk 1 above)
Comment (duckduckcodes) [reply backend/main.py:41]: true, i forgot to change, will do it right now

File: backend/main.py [hunk 5]
```diff
@@ -78,6 +83,30 @@ async def upload_files(files: List[UploadFile] = File(...)):
...
+    for f in files:
+        if f.startswith(file.file):
+            file_path = os.path.join(misc_dir, f)
+            os.remove(file_path)
+            
+            db.execute(text("DELETE FROM files WHERE file_name = :name"), {"name": file.file})
+            db.commit()
+            
+            # Introduce a delay to allow other processes or connections to access the database
+            time.sleep(0.1)  # Adjust the duration as needed
```
[the remark this answers is not in the document]
Comment (duckduckcodes) [reply backend/main.py:105]: i added this to avoid the db lock, we can maybe add multi threading later to process each file in a separate thread

File: frontend/src/renderer/App.tsx (hunk 4 above)
Comment (duckduckcodes) [reply frontend/src/renderer/App.tsx:81]: i didn't change anything, i added <Provider> to make the state accessible for the whole app (redux)

File: backend/main.py [hunk 6]
```diff
@@ -78,6 +83,29 @@ async def upload_files(files: List[UploadFile] = File(...)):
...
+    for f in files:
+        if f.startswith(file.document):
+            file_path = os.path.join(misc_dir, f)
+            os.remove(file_path)
+            
+            db.execute(text("DELETE FROM files WHERE file_name = :name"), {"name": file.document})
+            db.commit()
+            
+            # Introduce a delay to allow other processes or connections to access the database
+            time.sleep(0.1)  # Adjust the duration as needed
```
Comment (grumpyp) [inline backend/main.py:104]: The part of the deletion in the db can be outside of the loop

File: frontend/src/renderer/features/fileSplice.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
+
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { type } from 'os';
+import { FileFormat, FileSpliceState, RootState } from 'renderer/Data';
+
+const initialState: FileSpliceState = {
+    files: [],
+  };
+
```
Comment (grumpyp) [inline frontend/src/renderer/features/fileSplice.ts:10]: can we move this to a file where all the interfaces are? like the `index.ts` you introduced? 

Would be cool if you'd make it consistent anyway

File: frontend/src/renderer/pages/file-upload/components/Uploaded.tsx [hunk 8]
```diff
@@ -28,19 +34,62 @@ const useStyles = createStyles((theme) => ({
...
+      .then((response) => {
+        const fetchedFiles = response.data;
+        console.log(fetchedFiles);
+        if (fetchedFiles.length > 0) {
+          dispatch(addFile(fetchedFiles))
+          setFiles(fetchedFiles)
+        }
+      })
+      .catch((error) => {
+        console.log('Error fetching config:', error);
```
Comment (grumpyp) [inline frontend/src/renderer/pages/file-upload/components/Uploaded.tsx:63]: this log message has nothing to do with the behavior of the hook?
